ruby/lib/ci/queue/grind.rb

```diff
@@ -0,0 +1,26 @@
+require 'ci/queue/static'
+
+module CI
+  module Queue
+    class Grind < Static
+      class << self
+        def from_uri(uri, config)
+          new(uri.path, config)
+        end
+      end
+
+      def initialize(path, config)
+        io = path == '-' ? STDIN : ::File.open(path)
+
+        tests_to_run = io.each_line.map(&:strip).reject(&:empty?).to_set
+        tests = Minitest.loaded_tests.select do |test|
```

Comment:
You shouldn't access `Minitest` stuff in here.

There is 3 main section in ci queue:

  - `lib/minitest` the minitest integration
  - `lib/rspec` the rspec integration
  - `lib/ci` the abstract code that backs both.

Anyway, a Queue implementation should be initialized with string identifiers, the mapping to actual test framework objects is done later via `populate`.

Reply:
👍 thanks for the clarification. I changed the code appropriately.